### PR TITLE
add generator for genBinaryOutput and genBinaryInput

### PR DESCRIPTION
**Pull request by tomaszduda23**
Implement generator for genBinaryOutput and genBinaryInput which was mentioned here https://github.com/Koenkk/zigbee2mqtt/issues/23708

```js
const {deviceEndpoints, binary} = require('zigbee-herdsman-converters/lib/modernExtend');

const definition = {
    zigbeeModel: ['v1'],
    model: 'v1',
    vendor: 'esphome',
    description: 'Automatically generated definition',
    extend: [deviceEndpoints({"endpoints":{"1":1,"2":2,"3":3,"4":4,"5":5,"6":6,"7":7,"8":8}}), binary({"name":"zigbee1","cluster":"genBinaryOutput","attribute":"presentValue","reporting":{"attribute":"presentValue","min":"MIN","max":"MAX","change":1},"valueOn":["ON",1],"valueOff":["OFF",0],"description":"binary_output_1","access":"ALL","endpointName":"1"}), binary({"name":"zigbee2","cluster":"genBinaryOutput","attribute":"presentValue","reporting":{"attribute":"presentValue","min":"MIN","max":"MAX","change":1},"valueOn":["ON",1],"valueOff":["OFF",0],"description":"binary_output_2","access":"ALL","endpointName":"2"}), binary({"name":"zigbee3","cluster":"genBinaryOutput","attribute":"presentValue","reporting":{"attribute":"presentValue","min":"MIN","max":"MAX","change":1},"valueOn":["ON",1],"valueOff":["OFF",0],"description":"binary_output_3","access":"ALL","endpointName":"3"}), binary({"name":"zigbee4","cluster":"genBinaryOutput","attribute":"presentValue","reporting":{"attribute":"presentValue","min":"MIN","max":"MAX","change":1},"valueOn":["ON",1],"valueOff":["OFF",0],"description":"binary_output_4","access":"ALL","endpointName":"4"}), binary({"name":"name_test1","cluster":"genBinaryInput","attribute":"presentValue","reporting":{"attribute":"presentValue","min":"MIN","max":"MAX","change":1},"valueOn":["ON",1],"valueOff":["OFF",0],"description":"binary_input_5","access":"STATE_GET","endpointName":"5"}), binary({"name":"name_test2","cluster":"genBinaryInput","attribute":"presentValue","reporting":{"attribute":"presentValue","min":"MIN","max":"MAX","change":1},"valueOn":["ON",1],"valueOff":["OFF",0],"description":"binary_input_6","access":"STATE_GET","endpointName":"6"}), binary({"name":"name_test3","cluster":"genBinaryInput","attribute":"presentValue","reporting":{"attribute":"presentValue","min":"MIN","max":"MAX","change":1},"valueOn":["ON",1],"valueOff":["OFF",0],"description":"binary_input_7","access":"STATE_GET","endpointName":"7"}), binary({"name":"name_test4","cluster":"genBinaryInput","attribute":"presentValue","reporting":{"attribute":"presentValue","min":"MIN","max":"MAX","change":1},"valueOn":["ON",1],"valueOff":["OFF",0],"description":"binary_input_8","access":"STATE_GET","endpointName":"8"})],
    meta: {"multiEndpoint":true},
};

module.exports = definition;
```

### CODE REVIEW

File: src/lib/generateDefinition.ts [hunk 1]
```diff
@@ -334,3 +336,45 @@ async function extenderElectricityMeter(device: Zh.Device, endpoints: Zh.Endpoin
     }
     return [new Generator({extend: m.electricityMeter, args, source: `electricityMeter`})];
 }
+
+async function extenderBinaryInput(device: Zh.Device, endpoints: Zh.Endpoint[]): Promise<GeneratedExtend[]> {
+    const generated: GeneratedExtend[] = [];
+    for (const endpoint of endpoints) {
+        const source = 'binary_input';
```
Comment (Koenkk) [inline src/lib/generateDefinition.ts:343]:
```suggestion
        const source = 'binary';
```

I think this should be `binary`, but double check if the generate definition code is OK.

Comment (tomaszduda23) [reply src/lib/generateDefinition.ts:343]: It is possible to have both clusters at the same end point. Both would end up with the same name.

Comment (tomaszduda23) [reply src/lib/generateDefinition.ts:343]: you are right :)

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**